packages/block-library/src/post-content/index.php

```diff
@@ -14,6 +14,10 @@
  * @return string Returns the filtered post content of the current post.
  */
 function render_block_core_post_content( $attributes, $content, $block ) {
+	if ( ! in_the_loop() ) {
```

Comment:
You can place this below the early return `! isset( $block->context['postId'] )`.

Also should we use `wp_reset_postdata();` somewhere after calling `the_post`? It seems to behave okay with `Query` that includes `Post Content` as well, but would like your thoughts here.

Reply:
> You can place this below the early return `! isset( $block->context['postId'] )`.

Done.

> Also should we use `wp_reset_postdata();` somewhere after calling `the_post`? It seems to behave okay with `Query` that includes `Post Content` as well, but would like your thoughts here.

Not really necessary... Calling `the_post()` when the block is outside a loop will get the global `$post` so resetting to that won't make a difference. When the block is inside a loop, then `wp_reset_postdata()` runs at the end of the loop so we're covered again 👍